### DISCUSSION TITLE
Welcome to musiclm-pytorch Discussions!

### DISCUSSION THREAD

**Discussion by lucidrains** [lucidrains/musiclm-pytorch#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by aadityaverma** [comment]
excited! how do we get it to hallucinate enough that Shazam doesn't ~false positive it .

**Comment by albusdemens** [comment]
Did anyone manage to actually use the repo? Any tip is very welcome!

**Comment by onexdata** [comment]
Still trying to figure it out myself. The readme appears incomplete or for people more experienced with music ml.

So far I got this:

first do ```pip install musicml-pytorch```

Create a folder, download the two files from this github repo: https://github.com/facebookresearch/fairseq/tree/main/examples/hubert (choose hubert_base_ls960.pt and hubert_base_ls960_L9_km500.bin) and place them in a /hubert subfolder.

make a python file in the root and copy the code from the readme.md file (put in a single block)

*Note:* If you get the error ```AssertionError: Torch not compiled with CUDA enabled``` but you have a CUDA card (like an nVidia RTX2060, 2080, 4090, etc) and are on windows, do this:

```pip install torch===1.5.0 torchvision===0.6.0 -f https://download.pytorch.org/whl/torch_stable.html```

It will take a while because it will download a gigabyte of files.



This will get you at least past part 3 in the readme file, but I start getting all kinds of errors and can't get past them yet

**Reply by LWprogramming** [reply]
assuming that you meant `musiclm` instead of `musicml` :)

**Reply by mrfunkalot** [reply]
I've gotten this about this far:

I disabled cuda (because apparently version 12.1 isnt supported) by removing the .cuda() part of the end of the semantic_transformer function in the code block provided.

I also had to uninstall Python and reinstall an earlier version because of compatibility issues. 

I then installed SoundFile, and other dependencies that were needed, but I am now getting the error:

```
spectrogram yielded shape of (65, 86), but had to be cropped to (64, 80) to be patchified for transformer
..\Python38\site-packages\sklearn\base.py:318: UserWarning: Trying to unpickle estimator MiniBatchKMeans from version 0.24.0 when using version 1.2.2. This might lead to breaking code or invalid results. Use at your own risk. For more info please refer to:
https://scikit-learn.org/stable/model_persistence.html#security-maintainability-limitations
  warnings.warn(
Traceback (most recent call last):
  File ".\main.py", line 71, in <module>
    trainer = SemanticTransformerTrainer(
  File "<@beartype(audiolm_pytorch.trainer.SemanticTransformerTrainer.__init__) at 0x1ed6c0a44c0>", line 116, in __init__
  File "..\Python38\site-packages\audiolm_pytorch\trainer.py", line 592, in __init__
    self.ds = SoundDataset(
  File "<@beartype(audiolm_pytorch.data.SoundDataset.__init__) at 0x1ed6c093e50>", line 116, in __init__
  File "..\Python38\site-packages\audiolm_pytorch\data.py", line 46, in __init__
    assert path.exists(), 'folder does not exist'
AssertionError: folder does not exist
```
